ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -527,8 +541,43 @@ def _get_state(self, output: UnityRLOutputProto) -> AllBrainInfo:
             _data[brain_name] = BrainInfo.from_agent_proto(
                 self.worker_id, agent_info_list, self.brains[brain_name]
             )
+        self._parse_side_channel_message(output.side_channel)
         return _data
 
+    def _parse_side_channel_message(self, data: bytearray) -> None:
+        offset = 0
+        while offset < len(data):
+            try:
+                channel_type = struct.unpack("i", data[offset : offset + 4])[0]
```

Comment:
Maybe worth combining the reads here? especially since the tuple return value is unintuitive. I think something like
```
channel_type, message_len = struct.unpack("ii", data[offset : offset + 8])
offset += 8
```

Reply:
Also, it's a micro-optimization here but worth remembering when we get to shared memory - struct.unpack_from will probably be more efficient since it doesn't have to form the new bytearray from the slice.